src/main/java/taskitems/TaskManager.java

```diff
@@ -0,0 +1,236 @@
+            throw new IllegalInputException();
+        }
+        if (tasks.get(number - 1).isMarked()) {
+            System.out.println("Err, this task is already marked...");
+        } else {
+            tasks.get(number - 1).setMarked(true);
+            System.out.println("Nice! I've marked this task as done: ");
+            System.out.println(tasks.get(number - 1));
+        }
+        greet.printDecoration();
+        saveData();
+    }
+
+    public void unmarkTask(int number) throws IllegalInputException {
+        greet.printDecoration();
+        if(number > taskCount || number < 1){
+            throw new IllegalInputException();
+        }
+        if(!tasks.get(number - 1).isMarked()){
+            System.out.println("I cannot unmark something that was never marked...");
+        } else {
+            tasks.get(number - 1).setMarked(false);
+            System.out.println("Okay Boss! The following task has been unmarked: ");
+            System.out.println(tasks.get(number - 1));
+        }
+        greet.printDecoration();
+        saveData();
+    }
+
+    public void addToTasks(String taskName) throws IllegalInputException {
+        if (taskName.equals("")) {
+            throw new IllegalInputException();
+        }
+        greet.printDecoration();
+        tasks.add(new Todo(taskName));
+        System.out.println("added: " + tasks.get(taskCount));
+        taskCount++;
+        System.out.println("You now have " + taskCount + " tasks in the list.");
+        greet.printDecoration();
+        saveData();
+    }
+
+    public void addToTasks(String type, String taskName,String date){
+        greet.printDecoration();
+        if (type.equals("E")) {
+            tasks.add(new Event(taskName, date));
+        } else {
+            tasks.add(new Deadline(taskName, date));
+        }
+        System.out.println("added: " + tasks.get(taskCount));
+        taskCount++;
+        System.out.println("You now have " + taskCount + " tasks in the list.");
+        greet.printDecoration();
+        saveData();
+    }
+
+    private void addToTasks(String taskName, boolean isBackend) throws IllegalInputException {
+        if (taskName.equals("")) {
+            throw new IllegalInputException();
+        }
+        tasks.add(new Todo(taskName));
+        taskCount++;
+    }
+
+    private void addToTasks(String type, String taskName,String date, boolean isBackend){
+        if (type.equals("E")) {
+            tasks.add(new Event(taskName, date));
+        } else {
+            tasks.add(new Deadline(taskName, date));
+        }
+        taskCount++;
+    }
+
+    private void markTask(int number, boolean isBackend) throws IllegalInputException {
+        if (number > taskCount || number < 1) {
+            throw new IllegalInputException();
+        } else {
+            tasks.get(number - 1).setMarked(true);
+        }
+    }
+
+    public void loadData() throws FileNotFoundException {
```

Comment:
Another long method here, where you could apply [SLAP](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard).